pkg/compact/compact.go

```diff
@@ -968,6 +973,9 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 	cg.mtx.Lock()
 	defer cg.mtx.Unlock()
 
+	// the ctx here comes from bucketCompactor.compact and hence has the groups parent span in it
+	// hence, no need to create a new context/span here
```

Comment:
Do you still need to keep it?

Reply:
I usually keep these comments here. but I remove them in the upstream PR.